### ISSUE TITLE
Mask ideas

### ISSUE THREAD

**Issue by mpdehaan** [divamgupta/diffusionbee-stable-diffusion-ui#270]
Hi, this remains an awesome way to try out things in SD so thanks for all the hard work.

I noticed a few things:

- on "Image to Image" it would be nice to have the same mask slider feature that is available on "InPainting"
- on "InPainting" it would be nice to be able to change the prompt without having to redraw the mask
- on "Outpainting" it would be nice to be able to hit retry without having to redraw the mask again
- images from all tabs adding to history and  having "send to..." options would be great!

Basically merging these UIs to have the best features from each in each panel would have the best of all worlds, where you could draw the mask once and then play with the prompts, or go back and refine the mask without having to redraw the entire mask. 

Thoughts?

Once again, great app!

**Comment by eAi** [comment]
I’d also like to be able to load a mask from a file so I can create it in Photoshop as the tools there are much better than anything diffusion be has (or should have really)

**Comment by malaccan** [comment]
i would add to this to say that in my view, the 2 primary missing features for in-painting is:
a. the ability to save/load the mask once drawn, esp on complicated masks that takes ages to draw
b. undo/redo feature for drawing masks. currently, once a mistaken stroke is made, one has to redo the entire mask from scratch
thanks!
